src/test/lib/logical_query_plan/stored_table_node_test.cpp

```diff
@@ -438,16 +436,121 @@ TEST_F(StoredTableNodeTest, HasMatchingUniqueColumnCombination) {
   EXPECT_EQ(_stored_table_node->unique_column_combinations().size(), 1);
 
   // Negative test.
+  // Columns are empty.
+  EXPECT_THROW(_stored_table_node->has_matching_ucc({}), std::logic_error);
+
+  // There is no matching UCC.
   EXPECT_FALSE(_stored_table_node->has_matching_ucc({_b}));
   EXPECT_FALSE(_stored_table_node->has_matching_ucc({_c}));
   EXPECT_FALSE(_stored_table_node->has_matching_ucc({_b, _c}));
 
+  if constexpr (HYRISE_DEBUG) {
+    // Columns are not part of output_expressions() (i.e., pruned).
+    _stored_table_node->set_pruned_column_ids({ColumnID{0}});
+    EXPECT_THROW(_stored_table_node->has_matching_ucc({_a}), std::logic_error);
+  }
+
   // Test exact match.
+  _stored_table_node->set_pruned_column_ids({});
   EXPECT_TRUE(_stored_table_node->has_matching_ucc({_a}));
 
   // Test superset of column ids.
   EXPECT_TRUE(_stored_table_node->has_matching_ucc({_a, _b}));
   EXPECT_TRUE(_stored_table_node->has_matching_ucc({_a, _c}));
 }
 
+TEST_F(StoredTableNodeTest, OrderDependenciesSimple) {
+  EXPECT_TRUE(_stored_table_node->order_dependencies().empty());
+
+  // Create ODs from order constraints.
+  _table_a->add_soft_order_constraint({{ColumnID{0}}, {ColumnID{1}}});
+  _table_a->add_soft_order_constraint({{ColumnID{0}}, {ColumnID{2}}});
+  EXPECT_EQ(_table_a->soft_order_constraints().size(), 2);
+
+  const auto& order_dependencies = _stored_table_node->order_dependencies();
+  EXPECT_EQ(order_dependencies.size(), 2);
+  EXPECT_TRUE(order_dependencies.contains(OrderDependency{{_a}, {_b}}));
+  EXPECT_TRUE(order_dependencies.contains(OrderDependency{{_a}, {_c}}));
+}
+
+TEST_F(StoredTableNodeTest, OrderDependenciesPrunedColumns) {
+  // Discard ODs that involve pruned columns.
+  _table_a->add_soft_order_constraint({{ColumnID{0}}, {ColumnID{1}}});
+  _table_a->add_soft_order_constraint({{ColumnID{0}}, {ColumnID{2}}});
+  EXPECT_EQ(_table_a->soft_order_constraints().size(), 2);
+
+  _stored_table_node->set_pruned_column_ids({ColumnID{0}, ColumnID{2}});
+  const auto& order_dependencies = _stored_table_node->order_dependencies();
+  EXPECT_TRUE(order_dependencies.empty());
+}
+
+TEST_F(StoredTableNodeTest, OrderDependenciesTransitive) {
+  // Construct transitive ODs, but do not run into cycles. Furthermore, do not add ODs where LHS and RHS include the
```

Comment:
The description sounds strange. Rather "construct ... and check that no cycles are added.". "Check that no ODs where left and right ..." ??

Reply:
Would what the effect of the second sentence look like? I don't see how that is tested.

Reply:
So we have ODs `0 |-> 1`, `1 |-> 2` `2 |-> 0`. The `StoredTableNode` should generate all transitive ODs out of the input ODs. Thus, in a naïve implementation, the node would run circles creating these transitive ODs (starting with, e.g., `0 |-> 0`, and never ever stop. This is exactly what is tested here.